ci/ci-test-tasks/main.yml

```diff
@@ -6,78 +6,150 @@ parameters:
+          displayName: Process ${{ task }}
+          pool:
+            vmImage: windows-2022
+          steps:
+          # Start collect diagnostics
+          - powershell: ./ci/start-collect-diagnostics.ps1
+            displayName: Start collect diagnostics
 
-- stage: PreDelete
-  jobs:
-    - ${{ each task in split(parameters.tasks, ',') }}:    
-      - job: ${{ task }}_pre_delete
-        displayName: PreDelete (${{ task }})
-        pool:
-          vmImage: windows-2022
-        steps:
-        - template: ./delete-task.yml
-          parameters:
-            task: ${{ task }}
-            BuildSourceVersion: ${{ parameters.BuildSourceVersion }}
+          - template: ./build-init.yml
+            parameters:
+              BuildSourceVersion: ${{ parameters.BuildSourceVersion }}
 
-- stage: Upload
-  jobs:
-    - ${{ each task in split(parameters.tasks, ',') }}:
-      - job: ${{ task }}_upload_task
-        displayName: Upload (${{ task }})
-        pool:
-          vmImage: windows-2022
-        steps:
-        - template: ./upload-task.yml
-          parameters:
-            task: ${{ task }}
-            BuildSourceVersion: ${{ parameters.BuildSourceVersion }}
+          - template: ./build-task.yml
+            parameters:
+              task: ${{ task }}
+              patchVersion: true
+              version: '0.999.0'
 
-- stage: Test
-  jobs:
-    - ${{ each task in split(parameters.tasks, ',') }}:
-      - job: ${{ task }}_test_task
-        displayName: Test (${{ task }})
+          # Stop collect diagnostics
+          - powershell: ./ci/stop-collect-diagnostics.ps1
+            displayName: Stop collect diagnostics
+            condition: always()
+
+          - script: |
+              npm install -g tfx-cli@0.14
+            displayName: 'Npm install -g tfx-cli'
+
+          - script: |
+              tfx login --auth-type pat --service-url $(ADOUrl) --token $(ADOToken)
+            displayName: 'Tfx login'
+
+          - template: ./delete-task.yml
+            parameters:
+              task: ${{ task }}
+
+          - template: ./upload-task.yml
+            parameters:
+              task: ${{ task }}
+
+          - script: npm install axios@0.27.2
+            displayName: 'Npm install axios'
+
+          - script: |
+              echo "Task: ${{ task }}"
+              node ./ci/ci-test-tasks/test-and-verify.js $(ADOToken) $(ADOUrl) $(System.TeamProject) ${{ task }}
+            displayName: Run tests and verify results for ${{ task }}
+            failOnStderr: true
+
+          - template: ./delete-task.yml
+            parameters:
+              task: ${{ task }}
+
+          - ${{ if parameters.PushNugetPackage }}:
+            - template: ./push-nuget-init.yml
+
+            - script: |
+                cd $(Build.SourcesDirectory)\_package\nuget-packages\$(task)
+                push.cmd
+              displayName: Push Nuget package for $(task)
+
+    - ${{ if parameters.PublishHotfixArtifact }}:
```

Comment:
This part is redundant. We should never use PublishHotfixArtifact in this pipeline. It's a part of another pipeline where we make hotfixes. PublishHotfixArtifact is always false.

Reply:
Aha, that simplifies things a lot! Let me remove this part.